packages/netsuite-adapter/src/mapped_lists/utils.ts

```
@@ -0,0 +1,218 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Values, InstanceElement, isMapType, ObjectType, isField, isListType, Field,
+  MapType, isObjectType, createRefToElmWithValue, ListType, isContainerType, ElemID, Value,
+  BuiltinTypes,
+} from '@salto-io/adapter-api'
+import { naclCase, transformElement, TransformFunc, transformValues } from '@salto-io/adapter-utils'
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import _ from 'lodash'
+import { INDEX, LIST_MAPPED_BY_FIELD, SCRIPT_ID } from '../constants'
+import { listMappedByFieldMapping } from './mapping'
+import { CAPTURED_SERVICE_ID, captureServiceIdInfo } from '../service_id_info'
+
+const { awu } = collections.asynciterable
+
+const log = logger(module)
+
+export type MappedList = {
+  path: ElemID
+  value: Values
+}
+
+export const validateTypesFieldMapping = (elements: ObjectType[]): boolean => {
+  const typesMap = _.keyBy(elements, element => element.elemID.name)
+  return Object.entries(listMappedByFieldMapping).every(([typeName, fieldName]) => {
+    const type = typesMap[typeName]
+    if (type === undefined) {
+      log.error(`type '${typeName}' should have mapping field but '${typeName}' is not in elements`)
+      return false
```

I think that you can just throw the error from here and not from the calling function.